db/migrations/20190612135201_drop_title_from_movies.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+exports.up = (Knex) => {
+  return Knex.schema.table('movies', (table) => {
+    table.dropColumn('title');
+  })
+  .then(() => {
+    return Knex.raw('ALTER TABLE movies ADD CONSTRAINT movies_name_not_null CHECK (name IS NOT NULL) NOT VALID')
```

Comment:
Could you please simplify this to use `ALTER COLUMN name SET NOT NULL` and `DROP NOT NULL` (in rollback) rather than adding a check constraint?

Reply:
So, Miyagi defines this migration this way. I was curious why a `NOT NULL` constraint was used instead and did some digging around. Turns out adding a `NOT NULL` constraint on an existing column could lock your table for a longer time depending on the number of rows in that table preventing reads/writes, whereas adding a check constraint won't with the `NOT VALID` keyword. The following `VALIDATE` statement will scan the table with a `SHARE UPDATE EXCLUSIVE` lock which locks the table only for schema changes. So, while the `VALIDATE` statement is running, we can still read/write from the table.

Here's an example - https://medium.com/doctolib/adding-a-not-null-constraint-on-pg-faster-with-minimal-locking-38b2c00c4d1c

Let me know your thoughts!

Reply:
Cool makes sense. I assumed it was something regarding locking. Either way is fine, but in a lot of our services we do set NOT NULL constraints the way I suggested which could potentially cause issues?

Reply:
@paarig  We actually prefer to use `CHECK` constraints for not null when adding them to larger existing tables, for the reasons @shreyasj2006 pointed out! I think the only times we use `NOT NULL` are on creation typically or on smaller tables.